source/elements/oneMKL/source/domains/data_fitting/terms.rst

```diff
@@ -0,0 +1,324 @@
+
+       :math:`M = max_{i=1,\cdots,n-1} (\Delta_i)`,
+
+       :math:`m = min_{i=1,\cdots,n-1} (\Delta_i)`,
+
+       :math:`\Delta_i = x_{i+1} - x_i`
+   * - Vector-valued function of dimension ``p`` being fit
+     - :math:`f(x) = (f_1(x),\cdots, f_p(x))`.
+   * - A ``k``-order derivative of function ``f(x)`` at point ``t``
+     - :math:`f^{(k)}(t)`.
+   * - Function ``p`` agrees with function ``f`` at the points
+       :math:`\left\{ x_i \right\}_{i=1,\cdots,n}`.
+     - For every point :math:`\zeta` in sequence :math:`\left\{ x_i \right\}_{i=1,\cdots,n}`
+       that occurs ``m`` times, the equality :math:`p^{(i-1)}(\zeta) = f^{(i-1)}(\zeta)`
+       holds for all :math:`i=1,\cdots,m`.
+   * - The ``k``-th divided difference of function ``f`` at points :math:`x_i,..., x_{i+k}`.
+       This difference is the leading coefficient of the polynomial of order ``k+1``
+       that agrees with ``f`` at :math:`x_i,\cdots, x_{i+k}`.
+     - :math:`\left[  x_i,\cdots, x_{i + k} \right]f`.
+
+       In particular,
+
+       :math:`\left[ x_1 \right]f = f(x_1)`,
+
+       :math:`\left[  x_1, x_2 \right] f = (f(x_1) - f(x_2)) / (x_1 - x_2)`.
+
+Hints in the Data Fitting Component
+-----------------------------------
+
+The |IONE-MKL| Data Fitting component provides ways to specify some "hints"
+for partitions, function values, coefficients, interpolation sites.
+
+Partition Hints
+^^^^^^^^^^^^^^^
+
+The following are supported:
+
+ - Non-uniform.
+ - Quasi-uniform.
+ - Uniform.
+
+**Syntax**
+
+.. code:: cpp
+
+  enum class partition_hint {
+    non_uniform,
+    quasi_uniform,
+    uniform
+  };
+
+Function Values Hints
+^^^^^^^^^^^^^^^^^^^^^
+
+Let :math:`\left\{ x_i \right\}_{i=1,\cdots,nx}` is a partition,
+:math:`\left\{ f_j(x) \right\}_{j=1,\cdots,ny}` is a vector-valued function.
+Function values are stored in the one-dimensional array with ``nx * ny`` elements.
+2 different layouts are possible: row major and column major.
+
+- For row major layout function values are stored as the following:
+
+  Let :math:`\left\{ f_{j,i} \right\}_{j=1,\cdots,ny, i=1,\cdots,nx}`
+  is the function value that corresponds to the ``i-th`` partition point and the ``j-th`` function.
+
+- For column major:
+
+  Let :math:`\left\{ f_{i,j} \right\}_{i=1,\cdots,nx, j=1,\cdots,ny}`
+  is the function value that corresponds to the ``i-th`` partition point and the ``j-th`` function.
+
+The following hints are supported:
+
+ - Row major.
+ - Column major.
+
+**Syntax**
+
+.. code:: cpp
+
+  enum class function_hint {
+    row_major,
+    col_major
+  };
+
+Coefficients Hints
+^^^^^^^^^^^^^^^^^^
+
+Let :math:`\left\{ x_i \right\}_{i=1,\cdots,nx}` is a partition,
+:math:`\left\{ f_j(x) \right\}_{j=1,\cdots,ny}` is a vector-valued function.
+Let cubic spline should be constructed. It means that it requires 4 coefficients
+per each interpolation interval and function value.
+Cofficients are stored in the one-dimensional array with ``4 * (nx - 1) * ny`` elements.
```

Comment:
Let :math:`\left\{ x_i \right\}_{i=1,\cdots,nx}` be a partition of :math:`[a, b)` and
:math:`\mathbf{f}(x) = \left\{ f_j(x) \right\}_{j=1,\cdots,ny}` be a vector-valued function to be interpolated with a cubic spline on this partition.

To create a cubic spline, on each sub-interval in the partition, we require 4 coefficients per component :math:`f_j` of :mathbf{f}(x)`.  The coefficients are stored in a one-dimensional array with :math:` 4 (nx -1)(ny)` elements.